src/components/__tests__/change-view-test.jsx

```diff
@@ -0,0 +1,379 @@
+import {diffTypesFor} from '../../constants/diff-types';
+
+jest.mock('../../scripts/layered-storage');
+
+const mockChange = {
+  uuid: 'adc70e01-0723-4a28-a8e4-ca4b551ed2ae..0772dc83-7966-4881-8901-eceb051b9536',
+  uuid_from: 'adc70e01-0723-4a28-a8e4-ca4b551ed2ae',
+  uuid_to: '0772dc83-7966-4881-8901-eceb051b9536',
+  priority: null,
+  current_annotation: {},
+  created_at: null,
+  updated_at: null,
+  significance: null
+};
+
+// Note for testing relevant diffTypes:
+// the defaultDiffType (SIDE_BY_SIDE_RENDERED) is only relevant for the "text/html" media type
+// the relevant types for "text/*" are a subset of the relevant types for "text/html"
+// the relevant types for 'text/html' and "*/*" are mutually exclusive
+
+const defaultDiffType = 'SIDE_BY_SIDE_RENDERED';
+const diffTypeStorage = 'edgi.wm.ui.diff_type';
+
+describe('change-view', () => {
+  const mockApi = Object.assign(
+    Object.create(WebMonitoringDb.prototype),
+    {getChange: () => Promise.resolve(mockChange)}
+  );
+
+  describe('initial diffType', () => {
+    describe('when a diffType has been stored in layeredStorage', () => {
+      describe('when that diffType is relevant to the pages being compared', () => {
+        it('sets state.diffType to the stored value', () => {
+          const storedDiffType = 'CHANGES_ONLY_TEXT';
+          layeredStorage.getItem.mockReturnValue(storedDiffType);
+
+          const changeView = shallow(
+            <ChangeView
+              page={simplePage}
+              from={{content_type: 'text/html'}}
+              to={{content_type: 'text/html'}}
+              user={{email: 'me'}}
+            />,
+            {context: {api: mockApi}}
+          );
+
+          expect(changeView.state().diffType).toBe(storedDiffType);
+        });
+      });
+
+      describe('when that diffType is NOT relevant to the pages being compared', () => {
+        describe('when defaultDiffType (SIDE_BY_SIDE_RENDERED) is relevant to the pages being compared', () => {
+          it('sets state.diffType to SIDE_BY_SIDE_RENDERED', () => {
+            const storedDiffType = 'IRRELEVANT_DIFF_TYPE';
+            layeredStorage.getItem.mockReturnValue(storedDiffType);
+
+            const changeView = shallow(
+              <ChangeView
+                page={simplePage}
+                from={{content_type: 'text/html'}}
+                to={{content_type: 'text/html'}}
+                user={{email: 'me'}}
+              />,
+              {context: {api: mockApi}}
+            );
+
+            expect(changeView.state().diffType).toBe(defaultDiffType);
+          });
+        });
+
+        describe('when defaultDiffType (SIDE_BY_SIDE_RENDERED) is NOT relevant to the pages being compared', () => {
+          it('sets state.diffType to the first relevant diff type', () => {
+            const storedDiffType = 'IRRELEVANT_DIFF_TYPE';
+            layeredStorage.getItem.mockReturnValue(storedDiffType);
+
+            const mediaType = 'text/*';
```

Comment:
This should probably be an actual concrete type, like `text/xml` or `text/plain`. `diffTypesFor()` will parse the media type string and look for an exact match, then a generic type match (generic types are the part before the `/`), then an unknown match (`*/*`).